src/lib.rs

```diff
@@ -126,12 +126,74 @@ pub struct SelfEncryptor<'a> {
 }
 
 impl<'a> SelfEncryptor<'a> {
-  /// This is the only constructor, for encryptor object
+  /// This is the only constructor, for encryptor object.
   /// Each SelfEncryptor is used for a single file.
   /// The parameters are a DataMap and Storage.
-  /// If new file use DataMap::None as first parameter
+  /// If new file, use DataMap::None as first parameter.
   /// The get and put of Storage need to be implemented to
   /// allow the SelfEncryptor to store encrypted chunks and retrieve when necessary.
+  // # Example usage
+  // ```rust
+  // let mut se = SelfEncryptor::new(&mut my_storage as MyStorage, datamap::DataMap::None);
+  // ```
+  //
+
+/// # Example
+///
+/// SelfEncryptor Example
+///
+/// ```rust
+/// extern crate self_encryption
+/// extern crate rand;
+/// use use self_encryption::*;
+///
+/// fn main() {
```

Comment:
Good Leo nice to see you getting to grips with this, overwhelming perhaps but well worth the effort. 

Pair is simple in rust just use (thing1Type, thing2Type). So you can have Vec<(thinkg1type, thing2type)> or similar. So tuples everywhere more so than c++, but ADT's (enums) in rust also work great if you are only holding one type of many options (even with initialisers). 

Subtract should also be simple, if you post on slack what you are trying I will give you a hand, or post a playpen of it.